src/daft-micropartition/src/micropartition.rs

```diff
@@ -333,30 +345,41 @@ impl MicroPartition {
                 _,
                 FileFormatConfig::Parquet(ParquetSourceConfig {
                     coerce_int96_timestamp_unit,
-                    row_groups,
                 }),
                 StorageConfig::Native(cfg),
             ) => {
+                let uris = scan_task
+                    .sources
+                    .iter()
+                    .map(|s| s.get_path())
+                    .collect::<Vec<_>>();
                 let columns = scan_task
                     .columns
                     .as_ref()
                     .map(|cols| cols.iter().map(|s| s.as_str()).collect::<Vec<&str>>());
 
+                let row_groups = scan_task
```

Comment:
Is it maybe worth pulling `sources -> row_groups` calculation out into a helper/method?